superset/viz.py

```diff
@@ -575,15 +575,29 @@ def get_data(self, df: pd.DataFrame) -> VizData:
         the percent metrics have yet to be transformed.
         """
 
-        if not self.should_be_timeseries() and DTTM_ALIAS in df:
-            del df[DTTM_ALIAS]
-
+        non_percent_metric_columns = []
         # Transform the data frame to adhere to the UI ordering of the columns and
         # metrics whilst simultaneously computing the percentages (via normalization)
         # for the percent metrics.
-        non_percent_metric_columns = (
+
+        if not self.should_be_timeseries() and DTTM_ALIAS in df:
+            del df[DTTM_ALIAS]
+        elif self.should_be_timeseries() and DTTM_ALIAS in df:
+            non_percent_metric_columns.append(DTTM_ALIAS)
+
+        non_percent_metric_columns.extend(
             self.form_data.get("all_columns") or self.form_data.get("groupby") or []
-        ) + utils.get_metric_names(self.form_data.get("metrics") or [])
+        )
+
+        non_percent_metric_columns.extend(
+            utils.get_metric_names(self.form_data.get("metrics") or [])
+        )
```

Comment:
How are lines 588 - 594 different to what was previously on lines 584 - 586?

Reply:
I'm just extending them separately because it feels a bit more readable, rather than extending list + list.